samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManagerRestoreUtil.java

```diff
@@ -133,8 +135,8 @@ private static CompletableFuture<Map<TaskName, Checkpoint>> restoreAllTaskInstan
       Config config, Map<TaskName, TaskInstanceMetrics> taskInstanceMetrics, ExecutorService executor,
       File loggedStoreDir, Set<String> forceRestoreTask) {
 
-    Map<TaskName, CompletableFuture<Checkpoint>> newTaskCheckpoints = new ConcurrentHashMap<>();
-    List<Future<Void>> taskRestoreFutures = new ArrayList<>();
+    Map<TaskName, Checkpoint> newTaskCheckpoints = new ConcurrentHashMap<>();
+    Queue<Future<Void>> restoreAndCleanupFutures = new ConcurrentLinkedQueue<>();
```

Comment:
Why does this need to be a concurrent queue?

Reply:
Removed that. It was indeed unnecessary.